File: common/src/test/java/com/hartwig/actin/molecular/filter/SpecificGenesFilterTest.java
package com.hartwig.actin.molecular.filter;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import com.google.common.collect.Sets;

import org.junit.Test;

public class SpecificGenesFilterTest {

    @Test
    public void canFilterGenes() {
        SpecificGenesFilter filter = new SpecificGenesFilter(Sets.newHashSet("gene A"));

        assertTrue(filter.include("gene A"));
        assertFalse(filter.include("gene B"));
    }
}

File: common/src/test/kotlin/com/hartwig/actin/molecular/filter/SpecificGenesFilterTest.kt
package com.hartwig.actin.molecular.filter

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SpecificGenesFilterTest {

    @Test
    fun `Should filter specific genes`() {
        val filter = SpecificGenesFilter(setOf("gene A"))
        assertThat(filter.include("gene A")).isTrue
        assertThat(filter.include("gene B")).isFalse
    }
}